File: PSET6/sentimental-readability/readability.py
import string
from cs50 import get_string

text = get_string("Text: ")

l = 0
for i in text:
    if i.isalpha():
        l += 1

w = len(text.split())

s = 0
for m in text:
    if m == "." or m == "?" or m == "!":
        s += 1

words = w / 100
L = l / words
S = s / words

index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: PSET2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float letter(string text);
float word(string text);
float sentence(string text);

int main(void)
{
    string text = get_string("Text :");

    float letter_len = letter(text);
    float word_len = word(text);
    float sentence_len = sentence(text);

    float L = 100 * (letter_len / word_len);
    float S = 100 * (sentence_len / word_len);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

float letter(string text)
{
    int count = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            count++;
        }
    }
    return count;
}

float word(string text)
{
    int count = 1;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

float sentence(string text)
{
    int count = 0;
    int len = strlen(text);
    for (int i = 0; i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}